js/admin.js

```diff
@@ -0,0 +1,31 @@
+jQuery( document ).ready(function($) {
+	$('#check-status').click(function(){
+		$(".theme-check-report").empty();
+		themeCheckRunPHPCS();
+	});
+	function themeCheckRunPHPCS(){
+		$.ajax({
+			url: themeCheck.ajax_url,
```

Comment:
Can you check if this works. Before `url` add:

~~~js
xhr: function() {
	var xhr = new window.XMLHttpRequest();
	xhr.addEventListener( "progress", function (evt) {
		if ( evt.lengthComputable ) {
			var percentComplete = evt.loaded / evt.total;
			console.log('Percent completed: ' + (percentComplete * 100) );
		}
	}, false);
	return xhr;
},
~~~

This should show, in console the `percentComplete` changing. Which can then be easily used to toggle width of a progress bar.

I tried reproducing it locally, but for some reason it doesn't work for me.